extensions/mssql/src/schemaCompare/schemaCompareUtils.ts

```diff
@@ -295,13 +358,47 @@ export async function includeExcludeAllNodes(
     taskExecutionMode: TaskExecutionMode,
     payload: SchemaCompareReducers["includeExcludeAllNodes"],
     schemaCompareService: mssql.ISchemaCompareService,
+    logger?: Logger,
 ): Promise<mssql.SchemaCompareIncludeExcludeAllResult> {
+    logger?.info(
+        `[schemaCompareUtils] includeExcludeAllNodes called - operationId: ${operationId}, includeRequest: ${payload.includeRequest}`,
+    );
+    logger?.verbose(`[schemaCompareUtils] taskExecutionMode: ${taskExecutionMode}`);
+
+    const startTime = Date.now();
+    logger?.info(`[schemaCompareUtils] Calling schemaCompareService.includeExcludeAllNodes`);
+
     const result = await schemaCompareService.includeExcludeAllNodes(
         operationId,
         payload.includeRequest,
         taskExecutionMode,
     );
 
+    const elapsed = Date.now() - startTime;
+    logger?.info(
+        `[schemaCompareUtils] includeExcludeAllNodes service returned after ${elapsed}ms - success: ${result?.success}`,
+    );
+
+    if (result) {
+        logger?.info(
+            `[schemaCompareUtils] Result differences count: ${result.allIncludedOrExcludedDifferences?.length || 0}`,
+        );
+
+        if (result.errorMessage) {
+            logger?.error(`[schemaCompareUtils] Error message: ${result.errorMessage}`);
+        }
+
+        // Check for potential recursion issues based on elapsed time
+        if (elapsed > 30000) {
```

Comment:
If we aren't already we should emit a telemetry event for the total elapsed time so we can understand the distribution of times.  Event should probably also include the number of changes being included/excluded, too.